### ISSUE TITLE
Generated Database.ts is not compatible with TS 5.0 option verbatimModuleSyntax

### ISSUE THREAD

**Issue by thelinuxlich** [kristiandupont/kanel#436]
Since TS 5.0 there is a new option called [verbatimModuleSyntax](https://www.typescriptlang.org/tsconfig#verbatimModuleSyntax) which frameworks like SvelteKit set as true by default.

When this option is true, `export default Database` gives the following TS error:

`An 'export default' must reference a value when 'verbatimModuleSyntax' is enabled, but 'Database' only refers to a type.`

The solution is replacing this line:

```ts
// old
export default Database;
// new
export type { Database as default };
```

Workaround in .kanelrc.js
```ts
postRenderHooks: [
  (p, lines) => {
	if (p.endsWith("Database.ts")) {
		lines.pop();
		return [...lines, "export type { Database as default };"];
	}
	return lines;
  },
]
```

**Comment by kristiandupont** [comment]
Right, this is all something that I can't keep pushing much longer. I'll try to get around to looking into it soon!

**Comment by seivan** [comment]
There's another issue with `allowImportingTsExtensions`, lemme know if you want it in a separate ticket & pr.
The solutions are essentially the same, with allowing (or reading tsconfig) to understand what to do.

Current hack is
```ts
      postRenderHooks: [(_path, lines) => {
        return lines.map(l => l.includes("'./") ? l.replace("\';", '.ts\';') : l)
      }],
      ```

**Comment by alecmev** [comment]
`import type { ... }` is also preferred over `import { type ... }`, as it results in better elision (the import statements are completely removed, instead of being left behind as side effects). Combined with the above + ESM-friendly `.js` + named export of `Database`, this is the hook:

```ts
const verbatimModuleSyntaxHook: PostRenderHook = (_, lines) =>
  lines.map((x) =>
    x.startsWith('import { type default as') ?
      x
        .replace('import { type default as', 'import type')
        .replace('} ', '')
        .replace("';", ".js';")
    : x.startsWith('import {') ?
      x
        .replaceAll('type ', '')
        .replace('import {', 'import type {')
        .replace("';", ".js';")
    : x === 'export default Database;' ? 'export type { Database };'
    : x,
  );
```

**Comment by kristiandupont** [comment]
To subscribers here, the new [importsExtension](https://kristiandupont.github.io/kanel/configuring.html#importsextension) config field might be helpful. There is still some work to be done though.
